### ISSUE TITLE
CA1062: False positive when validating in a base method

### ISSUE THREAD

**Issue by joemey** [dotnet/roslyn-analyzers#4299]
### Analyzer

**Diagnostic ID**: [CA1062](https://docs.microsoft.com/visualstudio/code-quality/ca1062): `Validate arguments of public methods`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.0 (Latest)

### Describe the bug

Validating an input parameter in a base method is ignored in the overriden method

### Steps To Reproduce
```c#
    public class A
    {
        public virtual void Foo(string bar)
        {
            if (bar is null)
            {
                throw new ArgumentNullException(nameof(bar));
            }

            Console.Write($"A: {bar.Length}"); // <- does not trigger CA1062
        }
    }

    public class B : A
    {
        public override void Foo(string bar)
        {
            base.Foo(bar);
            Console.WriteLine($"B: {bar.Length}"); // <- will trigger CA1062
        }
    }
```
### Expected behavior

The analyzer should ignore the warning because there is a null check in the base method

### Actual behavior

the warning is generated

## Additional context

**Comment by mavasani** [comment]
This is by design. We do not perform interprocedural analysis for virtual method calls as it needs runtime vtable resolution to identify candidate method(s). You should instead decorate the parameter with `ValidatedNotNullAttribute` to indicate that the parameter is not null when the method returns from non-exceptional code paths.
